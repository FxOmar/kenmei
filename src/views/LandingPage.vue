<template lang="pug">
  #landing-page.font-sans.flex.flex-col.h-full
    content.flex-1
      hero-section(@signOnOpened="openSignOnWith($event)")
      features#features
      stats#stats
      call-to-action(@signOnOpened="openSignOnWith($event)")
    base-footer
    base-modal(
      :visible="signOnVisible"
      :loading="loading"
      @dialogClosed="signOnVisible = false"
      size="xs"
    )
      template(slot='body')
        component(
          :is='activeComponent'
          @componentChanged='activeComponent = $event'
          @signOnFinished='signOnVisible = false'
          @loading='loading = $event'
        )
</template>

<script>
  import HeroSection from '@/components/landing_page/LandingPageHeroSection';
  import Features from '@/components/landing_page/LandingPageFeatures';
  import Stats from '@/components/landing_page/LandingPageStats';
  import CallToAction from '@/components/landing_page/LandingPageCTA';
  import TheSignIn from '@/components/TheSignIn';
  import TheSignUp from '@/components/TheSignUp';
  import TheResetPassword from '@/components/TheResetPassword';

  export default {
    components: {
      HeroSection,
      Features,
      Stats,
      CallToAction,
      TheSignIn,
      TheSignUp,
      TheResetPassword,
    },
    data() {
      return {
        loading: false,
        signOnVisible: false,
        activeComponent: 'TheSignUp',
      };
    },
    methods: {
      openSignOnWith(comp) {
        this.activeComponent = comp;
        this.signOnVisible = true;
      },
    },
  };
</script>
