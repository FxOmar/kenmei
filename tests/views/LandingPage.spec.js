import { shallowMount } from '@vue/test-utils';
import LandingPage from '@/views/LandingPage.vue';

describe('LandingPage.vue', () => {
  it('renders the page', () => {
    const landingPage = shallowMount(LandingPage);
    expect(landingPage.element).toMatchSnapshot();
  });
});
