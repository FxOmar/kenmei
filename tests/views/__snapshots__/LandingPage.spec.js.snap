// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage.vue renders the page 1`] = `
<div
  class="font-sans"
  id="landing-page"
>
  <hero-section-stub />
  <features-stub
    id="features"
  />
  <stats-stub
    id="stats"
  />
  <call-to-action-stub />
  <base-footer-stub
    companyname="Studio Shogun, LTD"
    copyrightyear="2020"
    linkgroupone="[object Object]"
    linkgrouptwo="[object Object]"
    socials="[object Object]"
    subscribe="[object Object]"
  />
  <base-modal-stub
    size="xs"
  >
    <template>
      <thesignup-stub />
    </template>
  </base-modal-stub>
</div>
`;
